### PR TITLE
Update docs on CacheKeyGenerator

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
As suggested by @pracucci

### CODE REVIEW

File: pkg/frontend/querymiddleware/results_cache.go [hunk 1]
```diff
@@ -191,10 +191,14 @@ type CacheKeyGenerator interface {
 
 	// LabelValues should return a cache key for a label values request. The cache key does not need to contain the tenant ID.
 	// LabelValues can return ErrUnsupportedRequest, in which case the response won't be treated as an error, but the item will still not be cached.
+	// LabelValues should return a nil *GenericQueryCacheKey when it returns an error and
+	// should always return nil *GenericQueryCacheKey when the returned error is non-nil.
```
Comment (pracucci) [inline pkg/frontend/querymiddleware/results_cache.go:195]: I think you're saying the same thing twice.

Comment (dimitarvdimitrov) [reply pkg/frontend/querymiddleware/results_cache.go:195]: i am 🤦 
```suggestion
	// LabelValues should return a nil *GenericQueryCacheKey when it returns an error and
	// should always return non-nil *GenericQueryCacheKey when the returned error is nil.
```

File: pkg/frontend/querymiddleware/results_cache.go [hunk 2]
```diff
@@ -191,10 +191,14 @@ type CacheKeyGenerator interface {
...
 	LabelValues(ctx context.Context, path string, values url.Values) (*GenericQueryCacheKey, error)
 
 	// LabelValuesCardinality should return a cache key for a label values cardinality request. The cache key does not need to contain the tenant ID.
 	// LabelValuesCardinality can return ErrUnsupportedRequest, in which case the response won't be treated as an error, but the item will still not be cached.
+	// LabelValuesCardinality should return a nil *GenericQueryCacheKey when it returns an error and
+	// should always return nil *GenericQueryCacheKey when the returned error is non-nil.
```
Comment (dimitarvdimitrov) [inline pkg/frontend/querymiddleware/results_cache.go:201]:
```suggestion
	// should always return non-nil *GenericQueryCacheKey when the returned error is nil.
```

**Review by pracucci — APPROVED** [review]
Thanks!

**Pull request merged**